mypy/newsemanal/semanal.py

```diff
@@ -4508,3 +4507,12 @@ def names_modified_in_lvalue(lvalue: Lvalue) -> List[NameExpr]:
             result += names_modified_in_lvalue(item)
         return result
     return []
+
+
+def is_same_var_from_getattr(n1: Optional[SymbolNode], n2: Optional[SymbolNode]) -> bool:
+    """Do n1 and n2 refer to the same Var derived from module-level __getattr__?"""
+    return (isinstance(n1, Var)
+            and n1.from_module_getattr
+            and isinstance(n2, Var)
+            and n2.from_module_getattr
+            and n1.fullname() == n2.fullname())
```

Comment:
Can we just insert the newly created `Var` in the namespace we import from instead of using this function? Then we will not need also the new flag and all imports will actually point to the same `Var`.

Reply:
Modifying the ASTs of other modules makes things harder to reason about. For example, the other symbol table may already have been serialized, and now the serialized symbol table doesn't match what's in memory, which could be confusing.